packages/lib/src/core/core.ts

```diff
@@ -59,44 +65,23 @@ class Core {
         return Promise.resolve(this);
     }
 
-    /**
-     * Submit data to payments using the onSubmit event or the session flow if available
-     * @param data -
-     */
-    public submitPayment(data): void {
-        if (this.options.onSubmit) return this.options.onSubmit(data);
-
-        if (this.session) {
-            this.session
-                .submitPayment(data)
-                .then(response => {
-                    if (response.action) {
-                        if (this.options.onPaymentSubmitted) this.options.onPaymentSubmitted(response, this);
-                    } else {
-                        if (this.options.onPaymentCompleted) this.options.onPaymentCompleted(response, this);
-                    }
-                })
-                .catch(error => {
-                    if (this.options.onError) this.options.onError(error);
-                });
-        }
-    }
-
     /**
      * Submits details using onAdditionalDetails or the session flow if available
      * @param details -
      */
     public submitDetails(details): void {
-        if (this.options.onAdditionalDetails) return this.options.onAdditionalDetails(details);
+        if (this.options.onAdditionalDetails) {
+            return this.options.onAdditionalDetails(details);
+        }
 
         if (this.session) {
             this.session
                 .submitDetails(details)
                 .then(response => {
-                    if (this.options.onPaymentCompleted) this.options.onPaymentCompleted(response, this);
+                    this.options.onPaymentCompleted?.(response);
```

Comment:
If we remove `this`, it might break some merchants who are using the element ref. Same goes for `this.options.onError?.(error);`.

Reply:
I see, thanks for the explanation. It makes sense to omit it.